src/core/secure/storage.ts

```diff
@@ -0,0 +1,27 @@
+import AsyncStorage from '@react-native-community/async-storage';
+import { encrypt, decrypt } from './encrypt';
+
+const KEY_PREFIX = 'mw-';
+
+export const storeEncrypted = async (data: string, key: string, hash: string) => {
+    const encryptedData = await encrypt(data, hash);
+
+    try {
+        await AsyncStorage.setItem(`${KEY_PREFIX}${key}`, encryptedData);
+    } catch (e) {
+        // saving error
```

Comment:
return a `Promise.reject()` here, if `AsyncStorage.setItem()` is failing the function is returning no result.
Other approach would be to loose the `try...catch` and use it where `storeEncrypted()` is called.